src/STM32Ethernet.cpp

```diff
@@ -82,6 +85,27 @@ void EthernetClass::schedule(void)
   stm32_eth_scheduler();
 }
 
+void EthernetClass::macAddress(uint8_t *mac)
+{
+  // Read unique id
+  #if defined (STM32F2)
```

Comment:
Here it should be better to use `UID_BASE`
It's a common definition for all STM32 series defined by CMSIS.
`uint32_t baseUID = *(uint32_t *)UID_BASE;`